### DISCUSSION TITLE
Automatic grouping by package name?

### DISCUSSION THREAD

**Discussion by nielsbasjes** [renovatebot/renovate#10859] [Request Help] [answered]
If I want to get sets of updates that (to me) make sense to combine in one merge request I can make a config that contains something like this:
```json
  "packageRules": [
    {
      "matchPackagePrefixes": [
        "org.apache.logging",
        "org.slf4j"
      ],
      "groupSlug": "logging",
      "groupName": "logging",
      "labels": [
        "logging"
      ]
    },
```

If I do not configure anything I will get all updates as separate merge requests like this (assuming I do not have variable with the version number):

![image](https://user-images.githubusercontent.com/240096/125782072-049c74af-64fc-4c2e-a7a8-40d5f819492b.png)

Is there a generic way of having a "generic" setting where I say that all updates in the same maven package are to be grouped automatically without me having to specify each one in the config?

So I would like to see something like **Update nl.basjes.parse.useragent: to v6.0** without explicitly having this package name in the config.

**Comment by viceice** [comment]
There is no automatic way. you can try to match by source url (if it's available in maven registry), so you do not have to add all package names to the config

**Comment by rarkins** [comment, marked as answer]
This is something I hope to add in future. Right now we usually define branches and therefore group by package name but I think grouping by source URL by default would be better, and optionally allowing automatic grouping by scope/groupId instead
